src/pudl/workspace/datastore.py

```diff
@@ -1,715 +1,408 @@
+
+        with path.open("r") as f:
+            return yaml.load(f.read(), Loader=yaml.FullLoader)
+
+    # Remote resource retrieval
+
+    def is_remote(self, resource):
+        """
+        Determine whether a described resource is located on a remote server.
+
+        Args:
+            resource: dict, a resource descriptor from a frictionless data
+                      package
+        Returns:
+            boolean
+        """
+        return resource["path"][:8] == "https://" or \
+            resource["path"][:7] == "http://"
+
+    def passes_filters(self, resource, filters):
+        """
+        Test whether file metadata passes given filters.
+
+        Args:
+            resource: dict, a "resource" descriptior from a frictionless
+                      datapackage
+            filters: dict, pairs that must match in order for a
+                     resource to pass
+        Returns:
+            boolean, True if the resource parts pass the filters
+        """
+        for key, _ in filters.items():
+
+            part_val = resource["parts"].get(key, None)
+
+            if part_val != filters[key]:
+                self.logger.debug(
+                    "Filtered %s on %s: %s != %s", resource["name"], key,
+                    part_val, filters[key])
+                return False
+
+        return True
+
+    def download_resource(self, resource, directory):
+        """
+        Download a frictionless datapackage resource.
+
+        Args:
+            resource: dict, a remotely located resource descriptior from a
+                      frictionless datapackage
+            directory: the directory where the resource should be saved
+        Returns:
+            Path of the saved resource, or none on failure
+        """
+        response = requests.get(
+            resource["path"], params={"access_token": self.token})
+
+        if response.status_code >= 299:
+            msg = "Failed to download %s, %s" % (
+                resource["path"], response.text)
+            self.logger.error(msg)
+            raise RuntimeError(msg)
+
+        local_path = directory / resource["name"]
+
+        with local_path.open("wb") as f:
+            f.write(response.content)
+            self.logger.debug("Cached %s" % local_path)
+
+        return local_path
+
+    def _validate_dataset(self, dataset):
+        """
+        Make sure each resource on disc has the right md5sum.
+
+        Args:
+            dataset: name of a dataset
+        Returns:
+            Bool, True if local resources appear to be correct.
+        """
+        dpkg = self.datapackage_json(dataset)
+        ok = True
+
+        for r in dpkg["resources"]:
+
+            if self.is_remote(r):
+                self.logger.debug("%s not cached, skipping validation"
+                                  % r["path"])
+                continue
+
+            path = Path(r["path"])
+            with path.open("rb") as f:
+                m = hashlib.md5()
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of insecure MD2, MD4, MD5, or SHA1 hash function.](https://app.codacy.com/manual/zaneselvans/pudl/pullRequest?prid=5437498)